File: Practica_44.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.Scanner;

public class Practica_44 {

    public static String busqueda_dni(String fichero){
        Scanner scanner = new Scanner(System.in);
        System.out.println("Introduzca el dni: ");
        String dni = scanner.nextLine();
        String linea = null;
        try (FileReader fileReader = new FileReader(fichero);
        BufferedReader bufferedReader = new BufferedReader(fileReader);){
        while((linea=bufferedReader.readLine()) != null) {
            if (linea.substring(0, 9).equals(dni)){
                scanner.close();
                return linea;
            }
        }
        System.out.print("El dni introducido no existe");
        scanner.close();
        return linea;
        }
        catch (IOException e) {
            System.out.print("Error al leer el archivo");
            scanner.close();
            return linea;
        }
    }

    public static void imprimir_datos(String linea){
        String[] linea_c = linea.split(", ");
        System.out.println("nombre: " + linea_c[1]);
        System.out.println("apellidos: " + linea_c[2]);
        System.out.println("correo: " + linea_c[3]);
    } 

    public static void menu(){
        while (true){
            String fichero = "usuarios.txt";
            System.out.println("\n--- Opciones: ---");
            System.out.println("1. Buscar usuario por DNI");
            System.out.println("2. cerrar programa");
            Scanner scanner = new Scanner(System.in);
            System.out.println("Seleccione una opcion: ");
            String opcion = scanner.nextLine();
            if (opcion.equals("1")) {
                String linea = busqueda_dni(fichero);
                if (linea == null){
                    System.out.println("\nVuelva a elejir una opcion");
                }
                else {
                    imprimir_datos(linea);
                }
            }
            else if(opcion.equals("2")){
                scanner.close();
                return;
            }
        }
    }

    public static void main(String[] args){
        menu();
    }
}

File: Practica 44.py
# defino un metodo que pide un dni y busca la linea donde esta ese dni
def busqueda_dni(fichero):
    dni = str(input("Introduzca el dni que quieres buscar: "))
    for linea in fichero:
        # se establece un condicional que compara el dni introducido con el dni de cada linea del txt
        if dni == linea[:9]:
            return linea
    print("\nEl dni introducido no existe")
    
# defino un metodo para imprimir los datos de una linea
def imprimir_datos(linea):
    linea_c = linea.split(",")
    nombre = linea_c[1]
    apellidos = linea_c[2]
    correo = linea_c[3]
    print("nombre: " , nombre)
    print("apellidos: " , apellidos)
    print("correo: " , correo)
    
# defino el metodo del menu
def menu(fichero):
    while True:
        print("\n--- Opciones: ---")
        print("1. Buscar usuario por DNI")
        print("2. cerrar programa")
        opcion = input("Seleccione una opción: ")
        # si se elije la opcion 1 se llama al metodo de buscar dni y luego al metodo de imprimir los datos
        if opcion == '1':
            linea_actual = busqueda_dni(fichero)
            # se comprueba que la salida del metodo busqued_dni no sea null por haber introducido un dni inexistente
            if linea_actual != None:
                 imprimir_datos(linea_actual)
            else:
                print("\nVuelva a elejir una opcion")
        # si se elije la segunda opcion se cierra el programa
        elif opcion == '2':
            return
    


fichero = open('usuarios.txt' , 'r' , encoding='utf-8')
menu(fichero)
fichero.close()